### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Add Frictionless support to prepare for Py3/CKAN 2.10
- Add support for authenticating via config token
- Adjust dependencies and syntax to handle CKAN 2.10

### CODE REVIEW

File: .docker/scripts/init-ext.sh [hunk 1]
```diff
@@ -37,4 +44,7 @@ installed_name=$(grep '^\s*name=' setup.py |sed "s|[^']*'\([-a-zA-Z0-9]*\)'.*|\1
 # Validate that the extension was installed correctly.
 if ! pip list | grep "$installed_name" > /dev/null; then echo "Unable to find the extension in the list"; exit 1; fi
 
+if [ -d "$SRC_DIR/ckan/ckanext/activity" ]; then
```
Comment (duttonw) [inline .docker/scripts/init-ext.sh:47]: is this ckan 2.10 item?

File: ckanext/qa/model.py [hunk 2]
```diff
@@ -75,11 +75,10 @@ def create(cls, resource_id):
         c.resource_id = resource_id
 
         # Find the package_id for the resource.
-        q = model.Session.query(model.Package.id)
+        q = model.Session.query(model.Resource.package_id)
```
Comment (duttonw) [inline ckanext/qa/model.py:78]: is this more efficient?

**Review by duttonw — APPROVED** [review]

File: .docker/scripts/init-ext.sh (hunk 1 above)
Comment (ThrawnCA) [reply .docker/scripts/init-ext.sh:47]: Yes, the activity stream is being split from core and put into a plugin (but a plugin that is available out of the box, just needs to be enabled).

File: ckanext/qa/model.py (hunk 2 above)
Comment (ThrawnCA) [reply ckanext/qa/model.py:78]: It's more robust. In some cases attempting to load `model.Package` gives an error.

**Pull request merged**